src/Fixer/LanguageConstruct/NullableTypeDeclarationFixer.php

```diff
@@ -0,0 +1,335 @@
+    private const OPTION_SYNTAX_QUESTION_MARK = 'question_mark';
+
+    private int $candidateTokenKind;
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Nullable single type declaration should be standardised using configured syntax.',
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction bar(null|int \$value, null|\\Closure \$callable): void {}\n",
+                    new VersionSpecification(8_00_00)
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction baz(?int \$value, ?\\stdClass \$obj, ?array \$config): ?int {}\n",
+                    new VersionSpecification(8_00_00),
+                    ['syntax' => self::OPTION_SYNTAX_UNION]
+                ),
+                new VersionSpecificCodeSample(
+                    '<?php
+class ValueObject
+{
+    public null|string $name;
+    public ?int $count;
+    public null|bool $internal;
+    public null|\Closure $callback;
+}
+',
+                    new VersionSpecification(8_00_00),
+                    ['syntax' => self::OPTION_SYNTAX_QUESTION_MARK]
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return \PHP_VERSION_ID >= 8_00_00 && $tokens->isTokenKindFound($this->candidateTokenKind);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before OrderedTypesFixer, TypesSpacesFixer.
+     * Must run after NullableTypeDeclarationForDefaultNullValueFixer, SingleSpaceAroundConstructFixer.
+     */
+    public function getPriority(): int
+    {
+        return 2;
+    }
+
+    public function configure(array $configuration): void
+    {
+        parent::configure($configuration);
+
+        $this->candidateTokenKind = self::OPTION_SYNTAX_QUESTION_MARK === $this->configuration['syntax']
+            ? CT::T_TYPE_ALTERNATION // `|` -> `?`
+            : CT::T_NULLABLE_TYPE; // `?` -> `|`
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('syntax', 'Whether to use question mark (`?`) or explicit `null` union for nullable type.'))
+                ->setAllowedValues([self::OPTION_SYNTAX_UNION, self::OPTION_SYNTAX_QUESTION_MARK])
+                ->setDefault(self::OPTION_SYNTAX_QUESTION_MARK)
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $functionsAnalyzer = new FunctionsAnalyzer();
+
+        foreach (array_reverse($this->getElements($tokens), true) as $index => $type) {
+            if ('property' === $type) {
+                $this->normalizePropertyType($tokens, $index);
+
+                continue;
+            }
+
+            // fix return before arguments
```

Comment:
The fact return types are fixed before arguments is clear by just looking at the code below, if there are specific _reasons_ why it's like that, they should be included here. Otherwise the comment is superfluous.